### ISSUE TITLE
Stuck on "Wait for <bundle ID> to idle" when running on iOS Simulator with flavors

### ISSUE THREAD

**Issue by tbuczkowski** [leancodepl/patrol#2086]
### Steps to reproduce

1. Create a new Flutter project.
2. Setup Patrol in accordance to [Getting Started](https://patrol.leancode.co/getting-started)
3. Run `patrol test -t integration_test/example_test.dart`; test launches and completes successfully.
4. Setup flavors in the project in accordance to [Flutter's guide](https://docs.flutter.dev/deployment/flavors)
5. Add flavor to `patrol` section in pubspec.yaml.
6. Remove `FLUTTER_TARGET` path in xcode.
7. Run `patrol test -t integration_test/example_test.dart` again - patrol gets stuck on `Wait for <bundle ID> to idle`

### Actual results

Repo with reproducible example: https://github.com/tbuczkowski/patrol_test
Running `patrol test` from the initial commit of that repo works fine, but checking out anything after `Add flavors` commit results in the `Wait for <bundle ID> to idle` issue.

### Logs

<details>
<summary>Logs</summary>

```
Verbose mode enabled. More logs will be printed.
$ flutter --suppress-analytics --no-version-check pub deps --style=list
Received 1 test target(s)
Received test target: /Users/tymoteuszbuczkowski/Workspace/patrol_test/integration_test/example_test.dart
Generated entrypoint /Users/tymoteuszbuczkowski/Workspace/patrol_test/integration_test/test_bundle.dart with 1 bundled test(s)
Received Android flavor: prod
Received iOS flavor: prod
$ flutter --no-version-check --suppress-analytics devices --machine
No device specified, using the first one (iPhone 15 Pro Max)
Received 1 device(s) to run on
Received device: iPhone 15 Pro Max
Received 8 --dart-define(s) (0 custom, 8 internal)
Received internal --dart-define: PATROL_WAIT=0
Received internal --dart-define: PATROL_APP_BUNDLE_ID=com.example.patrol_test
Received internal --dart-define: PATROL_ANDROID_APP_NAME=My App
Received internal --dart-define: PATROL_IOS_APP_NAME=My App
Received internal --dart-define: INTEGRATION_TEST_SHOULD_REPORT_RESULTS_TO_NATIVE=false
Received internal --dart-define: PATROL_TEST_LABEL_ENABLED=true
Received internal --dart-define: PATROL_TEST_SERVER_PORT=8081
Received internal --dart-define: PATROL_APP_SERVER_PORT=8082
• Building app with entrypoint test_bundle.dart for iOS simulator (debug)...
$ flutter build ios --no-version-check --suppress-analytics --config-only --no-codesign --debug --simulator --flavor prod --target /Users/tymoteuszbuczkowski/Workspace/patrol_test/integration_test/test_bundle.dart --dart-define PATROL_WAIT=0 --dart-define PATROL_APP_BUNDLE_ID=com.example.patrol_test --dart-define PATROL_ANDROID_APP_NAME=My App --dart-define PATROL_IOS_APP_NAME=My App --dart-define INTEGRATION_TEST_SHOULD_REPORT_RESULTS_TO_NATIVE=false --dart-define PATROL_TEST_LABEL_ENABLED=true --dart-define PATROL_TEST_SERVER_PORT=8081 --dart-define PATROL_APP_SERVER_PORT=8082
        Building com.example.patrolTest for simulator (ios)...
$ xcodebuild build-for-testing -workspace Runner.xcworkspace -scheme prod -configuration Debug-prod -sdk iphonesimulator -destination generic/platform=iOS Simulator -quiet -derivedDataPath ../build/ios_integ OTHER_SWIFT_FLAGS=$(inherited) -D PATROL_ENABLED
✓ Completed building app with entrypoint test_bundle.dart for iOS simulator (12.6s)
Will uninstall apps before running tests
$ xcrun simctl uninstall A0F39D60-8132-4030-91D8-AA572D2DF3E9 com.example.patrol_test
$ xcrun simctl uninstall A0F39D60-8132-4030-91D8-AA572D2DF3E9 com.example.patrol_test.RunnerUITests.xctrunner
• Running app with entrypoint test_bundle.dart for iOS simulator on simulator iPhone 15 Pro Max...
$ xcodebuild -showsdks -json
Assuming SDK version 17.2 for iphonesimulator
Looking for .xctestrun matching prod_*iphonesimulator17.2*.xctestrun at /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_integ/Build/Products
Found 1 match(es), the first one will be used
Found /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_integ/Build/Products/prod_iphonesimulator17.2-arm64-x86_64.xctestrun
$ xcodebuild test-without-building -xctestrun /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_integ/Build/Products/prod_iphonesimulator17.2-arm64-x86_64.xctestrun -only-testing RunnerUITests/RunnerUITests -destination platform=iOS Simulator,name=iPhone 15 Pro Max -destination-timeout 1 -resultBundlePath /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_results_1706861394515.xcresult
        Command line invocation:
            /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild test-without-building -xctestrun /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_integ/Build/Products/prod_iphonesimulator17.2-arm64-x86_64.xctestrun -only-testing RunnerUITests/RunnerUITests -destination "platform=iOS Simulator,name=iPhone 15 Pro Max" -destination-timeout 1 -resultBundlePath /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_results_1706861394515.xcresult
        
        User defaults from command line:
            IDEBuildOperationResultBundlePath = /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_results_1706861394515.xcresult
            IDEPackageSupportUseBuiltinSCM = YES
        
        Writing result bundle at path:
                /Users/tymoteuszbuczkowski/Workspace/patrol_test/build/ios_results_1706861394515.xcresult
        
        --- xcodebuild: WARNING: Using the first of multiple matching destinations:
        { platform:iOS Simulator, id:A0F39D60-8132-4030-91D8-AA572D2DF3E9, OS:17.2, name:iPhone 15 Pro Max }
        { platform:iOS Simulator, id:A0F39D60-8132-4030-91D8-AA572D2DF3E9, OS:17.2, name:iPhone 15 Pro Max }
        2024-02-02 09:09:56.170524+0100 RunnerUITests-Runner[21576:30010127] [Default] Running tests...
            t =      nans Interface orientation changed to Portrait
        2024-02-02 09:09:57.405430+0100 RunnerUITests-Runner[21576:30010127] PatrolServer: INFO: PatrolServer constructor called
        2024-02-02 09:09:57.405494+0100 RunnerUITests-Runner[21576:30010127] PatrolServer: INFO: PATROL_ENABLED flag is defined
        2024-02-02 09:09:57.407158+0100 RunnerUITests-Runner[21576:30010127] PatrolServer: INFO: Starting server...
        2024-02-02 09:09:57.409360+0100 RunnerUITests-Runner[21576:30010127] PatrolServer: INFO: Server started on http://0.0.0.0:8081
        2024-02-02 09:09:57.409461+0100 RunnerUITests-Runner[21576:30010127] PatrolAppServiceClient: created, port: 8082
            t =      nans Checking existence of `"Allow" Button`
            t =      nans Open com.example.patrolTest
            t =      nans     Launch com.example.patrolTest
            t =      nans         Setting up automation session
            t =      nans         Wait for com.example.patrolTest to idle

```

</details>

### Patrol version

  patrol: ^3.5.0

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Patrol doctor:
Patrol CLI version: 2.6.2
Android: 
• Program adb found in /Users/tymoteuszbuczkowski/Library/Android/sdk/platform-tools/adb
• Env var $ANDROID_HOME set to /Users/tymoteuszbuczkowski/Library/Android/sdk
iOS / macOS: 
• Program xcodebuild found in /usr/bin/xcodebuild
• Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.16.9, on macOS 14.2.1 23C71 darwin-arm64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2023.1)
[✓] VS Code (version 1.85.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

</details>

**Comment by tbuczkowski** [comment]
I think this might be related: https://github.com/flutter/flutter/issues/112174. I have pretty much the same issue when running Flutter's integration_test package on iOS with flavors, but I was able to find a workaround - using `flutter drive` instead of `flutter test` command. With Patrol there's no other option than to use the patrol cli, is there?

**Comment by PiotrWpl** [comment]
@bartekpacia @mateuszwojtczak 
What is the official statement of you as Patrol dev team? Is there any way to make Patrol iOS tests work when a project has flavors? Or there is a dependency on Flutter SDK and we have to wait on the fix on the Flutter side here?

**Comment by bartekpacia** [comment]
Problems with Patrol with app using flavors have always been common. That's why [the example app](https://github.com/leancodepl/patrol/blob/master/packages/patrol/example) contains flavors. Clone and check it, maybe you'll find some difference that causes this bug.

(fyi I no longer work on Patrol)

**Comment by PiotrWpl** [comment]
Thank you @bartekpacia
your comment motivated me to compare my and the example app project settings, and I was able to run the tests on iOS too in the end!
I made a few changes, and clean-ups, but I think that the crucial thing was in my case it was probably to make my Configuration settings look like in the Getting Started
![image](https://github.com/leancodepl/patrol/assets/1515300/466d97b7-56af-4e79-9254-708f0628cafc)
instead of "the same as the Runner", because I had some strange values there (maybe because of `flutter_flavorizr`usage)

I hope @tbuczkowski that my comment will help to solve your issue too.

**Comment by yacineblr** [comment]
Do you have any news about this issue ?

**Comment by minhnguyenandpad** [comment]
Remove FLUTTER_TARGET get my cicd fail. Do you guys know why do we need to remove it?

**Comment by mateuszwojtczak** [comment]
It's hard to have any news on this issue. First, we need to ensure that this is reproducible. 

@piotruela AFAIK we do have flavors in our example app and it runs on the simulator, right?

If so, we should close this and improve docs. Otherwise, we should look into this as a potential blocker.

**Comment by mateuszwojtczak** [comment]
Okay, I talked with @piotruela and here's what we know:
1. Flavors work with iOS simulator in our example
2. We are not sure if our flavors are set up the same way as described in Flutter official docs
3. We will try to reproduce your issue with those steps:
- `flutter create`
- set up flavors as described in docs
- set up Patrol as described in docs
- check if everything works, if not - fix the docs

**Comment by pdenert** [comment]
So I investigate this issue in 2 ways.

1. First, I've checked that the patrol test runs correctly after performing the following steps:
- flutter create
- set up flavors as described in docs
- set up Patrol as described in docs

and without any additional changes it works like a charm.

2. As a second step, I've checked @tbuczkowski's repository and reproduced the problem. But the changes made in this commit: https://github.com/tbuczkowski/patrol_test/commit/b15446e8287561e6af5aa2f84e86b19c52eb67a0 are invalid. FLUTTER_TARGET shouldn't be set to `""`, it should be completely removed from the files. I've found that the Patrol FAQ gives this misleading advice:

> Open your XCode project, go to Runner target -> Build Settings, search for FLUTTER_TARGET and remove set path to targets for all configurations. 

which was wrong, because in XCode the user can't delete this, but only set it to empty string, which only sets the empty string.

With no changes from the above commit, the patrol tests pass.

As an action, I've removed this misleading note from the FAQ: https://github.com/leancodepl/patrol/pull/2217

tldr: patrol works with flutter flavors, we removed one misleading advice from FAQ
